### ISSUE TITLE
500 error with empty alert with link from specific website

### ISSUE THREAD

**Issue by littlepiggie** [nextcloud/cookbook#626]
**Description**
When I insert a particular link, the server errors and I get an alert without message.

**Reproduction**
1. Enter in "Download from URL" `https://www.ah.nl/allerhande/recept/R-R1185300/maanzaadbolletjes`.
2. Click button.
3. Empty popup, 500 internal server error in console.

**Expected behavior**
No 500 error. Clean message saying website is unsupported.

**Browser**
Firefox

**Versions**
Nextcloud server version: 20.0.1
Cookbook version: 0.8.1

**Comment by littlepiggie** [comment]
Note: the website uses the schema format, so it should work.

**Comment by cloud2018** [comment]
Perhaps the cookie banner on the website prevents the recipe from downloading correctly

**Comment by christianlupus** [comment]
The reason for this issue is that the website uses the context `http://www.schema.org/Recipe` in their microtypes. See here in line 308:
```
<article itemscope itemtype="http://www.schema.org/Recipe">
```
Our parser only looks for the `http://schema.org/Recipe` microtype context. Therefore the recipe is not found.
When implementing #553 this issue can be tackled as well.

**Comment by christianlupus** [comment]
At least now, there should be a better error message present.

**Comment by cvinhaes** [comment]
Hi there. Same here with XHR POST https://drive.cloud-io.network:2053/apps/cookbook/import [HTTP/2 500 Internal Server Error] and empty error message box in Firefox 87.0. I'm running Nextcloud 20.0.0. Same issue happens with many brazilian recipe sites, one example:
https://receitas.ig.com.br/guiadacozinha/2021-04-11/surpreenda-a-familia-com-a-receita-de-costelinha-de-porco-com-chutney-de-abacaxi.html

**Comment by christianlupus** [comment]
> https://receitas.ig.com.br/guiadacozinha/2021-04-11/surpreenda-a-familia-com-a-receita-de-costelinha-de-porco-com-chutney-de-abacaxi.html

This page does not support schema.org recipe meta data. Therefore, no import is possible ATM.

**Comment by cgudea** [comment]
It appears the recipes from SeriousEats have the same issue even though the scheme.org recipe metadata exists. Serious eats did work before

Example recipes:

* https://www.seriouseats.com/recipes/2015/02/grilled-lemongrass-coriander-marinated-tofu-vietnamese-sandwich-vegan-recipe.html
* https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html

Nextcloud: 20.0.9
Cookbook: 0.8.4

Let me know if you want more detailed logs, I just quickly grabbed this from the web admin page:

Warning | cookbook | libxml: Error 73 occurred 1 times while parsing https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html. Last time in line 2 and column 1: Couldn't find end of Start Tag or |   | 2021-04-18T14:53:10-0700
-- | -- | -- | -- | --
Warning | cookbook | libxml: Error 801 occurred 1 times while parsing https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html. Last time in line 2 and column 1: Tag or invalid |   | 2021-04-18T14:53:10-0700
Warning | cookbook | libxml: Error 68 occurred 1 times while parsing https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html. Last time in line 1 and column 67: error parsing attribute name |   | 2021-04-18T14:53:10-0700
Warning | cookbook | libxml: Error 23 occurred 1 times while parsing https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html. Last time in line 1 and column 47: htmlParseEntityRef: expecting ';' |   | 2021-04-18T14:53:10-0700
Warning | cookbook | libxml: Error 9 occurred 6 times while parsing https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html. Last time in line 1 and column 48: Invalid char in CDATA 0x12 |   | 2021-04-18T14:53:10-0700
Warning | cookbook | libxml: Error 73 occurred 1 times while parsing https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html. Last time in line 2 and column 1: Couldn't find end of Start Tag or |   | 2021-04-18T14:52:54-0700
Warning | cookbook | libxml: Error 801 occurred 1 times while parsing https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html. Last time in line 2 and column 1: Tag or invalid |   | 2021-04-18T14:52:54-0700
Warning | cookbook | libxml: Error 68 occurred 1 times while parsing https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html. Last time in line 1 and column 67: error parsing attribute name |   | 2021-04-18T14:52:54-0700
Warning | cookbook | libxml: Error 23 occurred 1 times while parsing https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html. Last time in line 1 and column 47: htmlParseEntityRef: expecting ';' |   | 2021-04-18T14:52:54-0700
Warning | cookbook | libxml: Error 9 occurred 6 times while parsing https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html. Last time in line 1 and column 48: Invalid char in CDATA 0x12 |   | 2021-04-18T14:52:54-0700
Warning | cookbook | libxml: Error 68 occurred 1 times while parsing https://www.seriouseats.com/recipes/2015/02/grilled-lemongrass-coriander-marinated-tofu-vietnamese-sandwich-vegan-recipe.html. Last time in line 1 and column 75: htmlParseStartTag: invalid element name |   | 2021-04-18T14:42:02-0700
Warning | cookbook | libxml: Error 23 occurred 1 times while parsing https://www.seriouseats.com/recipes/2015/02/grilled-lemongrass-coriander-marinated-tofu-vietnamese-sandwich-vegan-recipe.html. Last time in line 1 and column 44: htmlParseEntityRef: expecting ';' |   | 2021-04-18T14:42:02-0700
Warning | cookbook | libxml: Error 9 occurred 6 times while parsing https://www.seriouseats.com/recipes/2015/02/grilled-lemongrass-coriander-marinated-tofu-vietnamese-sandwich-vegan-recipe.html. Last time in line 1 and column 70: Invalid char in CDATA 0x1F |   | 2021-04-18T14:42:02-0700
Warning | cookbook | libxml: Error 23 occurred 3 times while parsing https://www.seriouseats.com/2015/02/how-to-make-vegan-vietnamese-grilled-tofu-lemongrass-coriander-sandwich-banh-mi.html. Last time in line 6 and column 28: htmlParseEntityRef: expecting ';' |   | 2021-04-18T14:41:09-0700
Warning | cookbook | libxml: Error 73 occurred 3 times while parsing https://www.seriouseats.com/2015/02/how-to-make-vegan-vietnamese-grilled-tofu-lemongrass-coriander-sandwich-banh-mi.html. Last time in line 6 and column 169: Couldn't find end of Start Tag p |   | 2021-04-18T14:41:09-0700
Warning | cookbook | libxml: Error 801 occurred 2 times while parsing https://www.seriouseats.com/2015/02/how-to-make-vegan-vietnamese-grilled-tofu-lemongrass-coriander-sandwich-banh-mi.html. Last time in line 5 and column 150: Tag kq invalid |   | 2021-04-18T14:41:09-0700
Warning | cookbook | libxml: Error 68 occurred 13 times while parsing https://www.seriouseats.com/2015/02/how-to-make-vegan-vietnamese-grilled-tofu-lemongrass-coriander-sandwich-banh-mi.html. Last time in line 6 and column 166: error parsing attribute name |   | 2021-04-18T14:41:09-0700
Warning | cookbook | libxml: Error 9 occurred 256 times while parsing https://www.seriouseats.com/2015/02/how-to-make-vegan-vietnamese-grilled-tofu-lemongrass-coriander-sandwich-banh-mi.html. Last time in line 6 and column 156: Invalid char in CDATA 0x1C |   | 2021-04-18T14:41:09-0700

**Comment by emichael** [comment]
I'm now getting 500 errors on every website I try. Same errors as @cgudea.

**Comment by christianlupus** [comment]
@cgudea I fear, your problem is not directly related to this issue.

> It appears the recipes from SeriousEats have the same issue even though the scheme.org recipe metadata exists. Serious eats did work before…
>     * https://www.seriouseats.com/recipes/2015/02/grilled-lemongrass-coriander-marinated-tofu-vietnamese-sandwich-vegan-recipe.html
> 
>     * https://www.seriouseats.com/recipes/2012/05/smoked-salmon-and-dill-tea-sandwiches.html

Yes, it might have worked. But currently, no schema.org metadata is present on their pages. So, no import will be possible (at least on those two example URLs).

The mentioned errors are present on the websites. We just get these errors when the app tries to import from a page that does not satisfy the HTML standard. They warn us that eventually, the parsed data is invalid. A manual inspection by me showed that no schema.org metadata is present. The errors are unrelated to this case.

@emichael if the websites you have problems with are known to having valid metadata, you can open a new issue (with example URLs), to check things.

This issue is not regarding any specific website that is not or only partly supported. The main issue is an empty error message in the case of a failed fetch. This empty message was fixed in #647 and I am closing this issue therefore now. I just forgot to do so in the past when merging.
